app/client/ui-modules/settings-page/components/ChangePasswordPopup.tsx

```diff
@@ -0,0 +1,185 @@
+"use client"
+import * as React from "react";
+import { z } from "zod";
+import { zodResolver } from "@hookform/resolvers/zod";
+import { useForm } from "react-hook-form";
+import { Accounts } from "meteor/accounts-base";
+import { useAppDispatch, useAppSelector } from "/app/client/store";
+import {
+  changePassword,
+  clearForm as clearChangePasswordForm,
+  selectChangePasswordForm,
+  setConfirmPassword,
+  setCurrentPassword,
+  setNewPassword,
+} from "/app/client/ui-modules/settings-page/state/reducers/change-password-slice";
+
+import { Dialog, DialogContent, DialogHeader, DialogTitle, DialogTrigger, DialogFooter, DialogClose } from "/app/client/ui-modules/theming-shadcn/Dialog";
+import { Button } from "/app/client/ui-modules/theming-shadcn/Button";
+import { Input } from "/app/client/ui-modules/theming-shadcn/Input";
+import { Toast } from "/app/client/ui-modules/settings-page/components/Toast";
+import {
+  Form,
+  FormControl,
+  FormDescription,
+  FormField,
+  FormItem,
+  FormLabel,
+  FormMessage,
+} from "/app/client/ui-modules/theming-shadcn/Form";
+
+const schema = z
+  .object({
+    currentPassword: z.string().min(1, "Current password is required"),
+    newPassword: z
+      .string()
+      .min(8, "New password must be at least 8 characters"),
+    confirmPassword: z.string().min(1, "Please confirm your new password"),
+  })
+  .refine((data) => data.newPassword === data.confirmPassword, {
+    message: "Passwords do not match",
+    path: ["confirmPassword"],
+  })
+  .refine((data) => data.newPassword !== data.currentPassword, {
+    message: "New password must be different from current password",
+    path: ["newPassword"],
+  });
+
+type FormValues = z.infer<typeof schema>;
+
+export function ChangePasswordPopup(): React.JSX.Element {
+  const [open, setOpen] = React.useState(false);
+  const dispatch = useAppDispatch();
+  const changePasswordState = useAppSelector(selectChangePasswordForm);
+
+  const form = useForm<FormValues>({
+    resolver: zodResolver(schema),
+    defaultValues: {
+      currentPassword: "",
+      newPassword: "",
+      confirmPassword: "",
+    },
+    mode: "onSubmit",
+  });
+
+  async function onSubmit(values: FormValues): Promise<void> {
+    // Sync form values into Redux slice for unified auth flows
+    dispatch(setCurrentPassword(values.currentPassword));
+    dispatch(setNewPassword(values.newPassword));
+    dispatch(setConfirmPassword(values.confirmPassword));
+
+    try {
+      // Close popup immediately on submit as requested
+      setOpen(false);
+      await dispatch(changePassword()).unwrap();
+      Toast({
+        title: "Password updated",
+        content: "Your password has been changed successfully.",
+      });
+      form.reset();
+      dispatch(clearChangePasswordForm());
+    } catch (err: any) {
+      Toast({
+        title: "Password change failed",
+        content: err || changePasswordState.error || "Unable to change password.",
```

Comment:
Very good to think og feedback but they are not really working. 
<img width="433" height="59" alt="Screenshot 2025-09-15 at 6 32 00 PM" src="https://github.com/user-attachments/assets/e6ade1b0-b944-4be4-845c-1527310238bd" />